src/typeahead/typeahead.ts

```diff
@@ -370,4 +377,13 @@ export class NgbTypeahead implements ControlValueAccessor,
     }
     this._subscription = null;
   }
+
+  private _callOnChange(value: any) {
+    this._model = value;
+    this._onChange(value);
+  }
+
+  private _inputDifferentFromModel(): boolean {
+    return this._formatItemForInput(this._model) !== this._elementRef.nativeElement.value;
```

Comment:
I don't think this is right since:
* several model values can lead to the same formatted string;
* we are reading from the DOM

We should probably listen to model changes instead to know if it was changed or not.